r2/r2/controllers/api.py

```diff
@@ -1062,7 +1062,8 @@ def POST_friend(self, form, jquery, friend,
         else:
             permissions = None
 
-        if type == "moderator_invite" and container.is_moderator(friend):
+        if (type in ("moderator_invite", "moderator") and
+           container.is_moderator(friend)):
```

Comment:
Style nit: wrong indentation, should be 8 spaces in from the `if` line so it looks like:

```
if (type in ("moderator_invite", "moderator") and
        container.is_moderator(friend)):
    c.errors.add(errors.ALREADY_MODERATOR, field="name")
```